Content.Server/Station/Systems/StationSpawningSystem.cs

```diff
@@ -205,6 +202,28 @@ public override void Initialize()
             EquipRoleLoadout(entity.Value, loadout, roleProto);
         }
 
+        if (loadout != null && roleProto != null)
+        {
+            foreach (var group in loadout.SelectedLoadouts.OrderBy(x => roleProto.Groups.FindIndex(e => e == x.Key)))
+            {
+                foreach (var items in group.Value)
+                {
+                    if (!_prototypeManager.TryIndex(items.Prototype, out var loadoutProto))
+                    {
+                        continue;
+                    }
+
+                    if (loadoutProto.Entity != null)
+                    {
+                        var newEntity = SpawnEntity(loadoutProto.Entity, coordinates, job);
+                        EntityManager.DeleteEntity(entity); //entity isnt deleted before as the loadout is on this entity
+                        entity = newEntity;
+                        return entity.Value;
+                    }
+                }
+            }
+        }
+
```

Comment:
This code's a bit of a mess.

You should just be checking for the entity up top and just for loadouts rather than equipping the loadout and suddenly bulldozing the entity entirely.